hexkit/inject.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Utilities for dependency injection based on the dependency_injector framework.
+
```

Comment:
You should state the main purpose of the module here, something like:

"This module contains a helper function that automatically selects the suitable "Provider" class for the dependency_injector framework and makes the "Container" class usable as async context manager."

Reply:
Done.